### DISCUSSION TITLE
Authentication with db

### DISCUSSION THREAD

**Discussion by quick007** [nextauthjs/next-auth#4060] [Help] [answered]
Hi all,
I want to store information about the user using redis. I currently have authentication set up, but I'm not sure how to tie my database info with their token. I assume there's an actual way to do this correctly?

**Comment by balazsorban44** [comment, marked as answer]
Hi, we have an Upstash Redis Adapter https://next-auth.js.org/adapters/upstash-redis

You could use that as a starting point if you don't need the Upstash part. Here is the source code:

https://github.com/nextauthjs/next-auth/blob/main/packages/adapter-upstash-redis/src/index.ts
